runtime/buffer.h

```diff
@@ -437,7 +427,10 @@ void for_each_index(const raw_buffer& buf, const F& f) {
 // This function attempts to be efficient to support production quality implementations of callbacks.
 template <typename F>
 void for_each_contiguous_slice(const raw_buffer& buf, const F& f) {
-  internal::for_each_contiguous_slice(buf.base, buf.dims, buf.rank - 1, buf.elem_size, f);
+  internal::for_each_contiguous_slice_dim* dims = SLINKY_ALLOCA(internal::for_each_contiguous_slice_dim, buf.rank + 1);
```

Comment:
It's not immediately obvious (to me) why the +1 is needed here; a comment is appropriate